src/test/java/com/google/planningworks/cdap/plugins/sink/AnaplanSinkTest.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.planningworks.cdap.plugins.sink;
+
+import static com.anaplan.client.AnaplanService.NAME_AUTH_SERVICE_LOCATION;
+import static com.anaplan.client.AnaplanService.NAME_MODEL_ID;
+import static com.anaplan.client.AnaplanService.NAME_PASSWORD;
+import static com.anaplan.client.AnaplanService.NAME_SERVER_FILE_NAME;
+import static com.anaplan.client.AnaplanService.NAME_SERVICE_LOCATION;
+import static com.anaplan.client.AnaplanService.NAME_USERNAME;
+import static com.anaplan.client.AnaplanService.NAME_WORKSPACE_ID;
+
+import com.google.planningworks.cdap.plugins.base.AnaplanPluginConfig;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.api.data.schema.Schema.LogicalType;
+import io.cdap.cdap.etl.mock.validation.MockFailureCollector;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.internal.util.reflection.FieldSetter;
+
+@RunWith(JUnit4.class)
+public class AnaplanSinkTest {
+
+  AnaplanSinkConfig config;
+  Schema schema;
+  MockFailureCollector failureCollector;
+
+  @Before
+  public void setUp() throws Exception {
+    failureCollector = new MockFailureCollector("AnaplanSinkTest");
+    config = getConfig();
+    schema = getSupportedSchema();
+  }
+
+  @Test
+  public void testParameterValidationPass() {
+    config.validate(failureCollector, schema);
+    Assert.assertEquals(/*expected =*/ 0, failureCollector.getValidationFailures().size());
+  }
+
+  @Test
+  public void testUnsupportedSchema() {
+    schema = getUnsupportedSchema();
+    config.validate(failureCollector, schema);
+    Assert.assertEquals(/*expected =*/ 5, failureCollector.getValidationFailures().size());
```

Comment:
better to build the set of fields that were incorrect from the ValidationFailures collected by the failureCollector and check that it's equal to the expected set of bad fields.

Reply:
Thanks! Updated!